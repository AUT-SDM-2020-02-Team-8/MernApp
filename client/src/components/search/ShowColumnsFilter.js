import React, { Component } from "react"
import { Menu, Button, MenuItem, FormControl, FormLabel, FormGroup } from "@material-ui/core"
import ShowColumnCheckbox from './ShowColumnCheckbox'

class ShowColumnsFilter extends Component {
  constructor(props) {
    super()
    this.state = {
      anchorEl: null,
      chosenColumns: Object.keys(props.headerMapping)
    }
    this.handleClick = this.handleClick.bind(this)
    this.handleClose = this.handleClose.bind(this)
    this.toggleColumn = this.toggleColumn.bind(this)
  }
  handleClick(event) {
    this.setState({ anchorEl: event.target })
  }

  handleClose() {
    this.setState({ anchorEl: null })
    this.props.showColumns(this.state.chosenColumns)
  }

  toggleColumn(event) {
    let chosenColumns = this.state.chosenColumns
    if (event.target.checked) {
      chosenColumns.push(event.target.name)
    }
    else {
      chosenColumns = this.state.chosenColumns.filter(item => item !== event.target.name)
    }
    this.setState({ chosenColumns: chosenColumns })
  }

  renderCheckBoxes() {
    let checkBoxes = []
    for (const [key, value] of Object.entries(this.props.headerMapping)) {
      checkBoxes.push(<ShowColumnCheckbox
        key={key}
        name={key}
        label={value}
        chosenColumns={this.state.chosenColumns}
        handleChange={this.toggleColumn}
      />)
    }
    return checkBoxes
  }

  render() {
    return (
      <div>
        <Button onClick={this.handleClick}>
          Show columns
        </Button>
        <Menu
          id="columnFilterBox"
          anchorEl={this.state.anchorEl}
          keepMounted
          open={Boolean(this.state.anchorEl)}
          onClose={this.handleClose}
        >
          <MenuItem>
            <FormControl component="fieldset">
              <FormLabel component="legend">Choose columns to show</FormLabel>
              <FormGroup>{this.renderCheckBoxes()}</FormGroup>
            </FormControl>
          </MenuItem>
        </Menu>
      </div>
    )
  }
}

export default ShowColumnsFilter