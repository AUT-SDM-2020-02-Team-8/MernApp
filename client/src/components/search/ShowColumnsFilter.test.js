import React from 'react';
import { shallow } from 'enzyme';
import ShowColumnsFilter from './ShowColumnsFilter';

test('renders homepage', () => {
  const headerMapping = {
    title: 'Title',
    author: 'Author',
    year: 'Year',
    recordType: 'Type',
    journal: 'Journal',
    publisher: 'Publisher',
    sePractice: 'SE Practice',
    claims: 'Claims'
  }
  const wrapper = shallow(<ShowColumnsFilter headerMapping={headerMapping} showColumns={Object.keys(headerMapping)}/>);
  expect(wrapper).toMatchSnapshot();
});
