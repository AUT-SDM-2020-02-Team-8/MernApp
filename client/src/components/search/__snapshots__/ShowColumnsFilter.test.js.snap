// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage 1`] = `
<div>
  <WithStyles(ForwardRef(Button))
    onClick={[Function]}
  >
    Show columns
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Menu))
    anchorEl={null}
    id="columnFilterBox"
    keepMounted={true}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(FormControl))
        component="fieldset"
      >
        <WithStyles(ForwardRef(FormLabel))
          component="legend"
        >
          Choose columns to show
        </WithStyles(ForwardRef(FormLabel))>
        <WithStyles(ForwardRef(FormGroup))>
          <ShowColumnCheckbox
            chosenColumns={
              Array [
                "levelOfSupport",
                "title",
                "author",
                "year",
                "recordType",
                "journal",
                "publisher",
                "sePractice",
                "claims",
              ]
            }
            handleChange={[Function]}
            key="levelOfSupport"
            label="Level of Support"
            name="levelOfSupport"
          />
          <ShowColumnCheckbox
            chosenColumns={
              Array [
                "levelOfSupport",
                "title",
                "author",
                "year",
                "recordType",
                "journal",
                "publisher",
                "sePractice",
                "claims",
              ]
            }
            handleChange={[Function]}
            key="title"
            label="Title"
            name="title"
          />
          <ShowColumnCheckbox
            chosenColumns={
              Array [
                "levelOfSupport",
                "title",
                "author",
                "year",
                "recordType",
                "journal",
                "publisher",
                "sePractice",
                "claims",
              ]
            }
            handleChange={[Function]}
            key="author"
            label="Author"
            name="author"
          />
          <ShowColumnCheckbox
            chosenColumns={
              Array [
                "levelOfSupport",
                "title",
                "author",
                "year",
                "recordType",
                "journal",
                "publisher",
                "sePractice",
                "claims",
              ]
            }
            handleChange={[Function]}
            key="year"
            label="Year"
            name="year"
          />
          <ShowColumnCheckbox
            chosenColumns={
              Array [
                "levelOfSupport",
                "title",
                "author",
                "year",
                "recordType",
                "journal",
                "publisher",
                "sePractice",
                "claims",
              ]
            }
            handleChange={[Function]}
            key="recordType"
            label="Type"
            name="recordType"
          />
          <ShowColumnCheckbox
            chosenColumns={
              Array [
                "levelOfSupport",
                "title",
                "author",
                "year",
                "recordType",
                "journal",
                "publisher",
                "sePractice",
                "claims",
              ]
            }
            handleChange={[Function]}
            key="journal"
            label="Journal"
            name="journal"
          />
          <ShowColumnCheckbox
            chosenColumns={
              Array [
                "levelOfSupport",
                "title",
                "author",
                "year",
                "recordType",
                "journal",
                "publisher",
                "sePractice",
                "claims",
              ]
            }
            handleChange={[Function]}
            key="publisher"
            label="Publisher"
            name="publisher"
          />
          <ShowColumnCheckbox
            chosenColumns={
              Array [
                "levelOfSupport",
                "title",
                "author",
                "year",
                "recordType",
                "journal",
                "publisher",
                "sePractice",
                "claims",
              ]
            }
            handleChange={[Function]}
            key="sePractice"
            label="SE Practice"
            name="sePractice"
          />
          <ShowColumnCheckbox
            chosenColumns={
              Array [
                "levelOfSupport",
                "title",
                "author",
                "year",
                "recordType",
                "journal",
                "publisher",
                "sePractice",
                "claims",
              ]
            }
            handleChange={[Function]}
            key="claims"
            label="Claims"
            name="claims"
          />
        </WithStyles(ForwardRef(FormGroup))>
      </WithStyles(ForwardRef(FormControl))>
    </WithStyles(ForwardRef(MenuItem))>
  </WithStyles(ForwardRef(Menu))>
</div>
`;
